src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -250,6 +288,14 @@ static ucs_status_t uct_cuda_copy_mem_free(uct_md_h md, uct_mem_h memh)
 
 static void uct_cuda_copy_md_close(uct_md_h uct_md) {
     uct_cuda_copy_md_t *md = ucs_derived_of(uct_md, uct_cuda_copy_md_t);
+    CUdevice dev;
+
+    if (md->cuda_ctx_retained &&
+        (UCT_CUDADRV_FUNC_LOG_ERR(cuDeviceGet(&dev, 0)) == UCS_OK) &&
+        (UCT_CUDADRV_FUNC_LOG_ERR(cuDevicePrimaryCtxRelease(dev)) == UCS_OK)) {
+        md->cuda_ctx_retained = 0;
+        ucs_debug("released the primary context on the gpu %d", dev);
+    }
```

Comment:
looks a bit strange:
1. why is it needed to clear `md->cuda_ctx_retained` right before freeing this memory?
2. do we want to print something if we failed to release the context?

Reply:
1. removed
2. we print error from CUDA driver API calls.